hw/top_earlgrey/data/chip_testplan.hjson

```diff
@@ -97,56 +97,104 @@
     // SPI_DEVICE (pre-verified IP) integration tests:
     {
       name: chip_spi_device_tx_rx
-      desc: '''Verify the transmission of data on the chip's SPI device port.
+      desc: '''Verify the transmission of data on the chip's SPI device port in firmware mode with
```

Comment:
to be honest with you, once boot strap is moved to spi eeprom protocol, i'm not even sure who will be the user of this mode anymore.  Maybe it'll make sense for us to capture the boot rom code as a test, and always use it to test this mode. 

@eunchan on another note, Don mentioned to me that SPI TPM is still on the table, if that's the case, should we add an explicit test case also?

Reply:
> to be honest with you, once boot strap is moved to spi eeprom protocol, i'm not even sure who will be the user of this mode anymore. Maybe it'll make sense for us to capture the boot rom code as a test, and always use it to test this mode.

could you explain how to "capture the boot rom code as a test" to verify the FW mode? Is it to load the boot rom image using FW mode?

Reply:
yeah because in the future, the actual boot rom will use a different protocol, so the existing boot strap test will no longer test this mode.

So i'm thinking we could move the existing rom code into a test, and use that to test this mode maybe... it's just a thought. Maybe there are much simpler ways to do the test also..

Reply:
As discussed, current approach that uses random data and checks received data are correct, is simpler.
Added a TODO to consider to use a real use case to verify this FW mode once we know how to use it.